design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,368 @@
+    end
+```
+
+## Rationale
+
+[A discussion of alternate approaches that have been considered and the trade
+offs, advantages, and disadvantages of the chosen approach.]
+
+## Affected components and Teams
+
+Edge Infrastructure Manager:
+- **Inventory (inv)**: changes to the schema 
+  1. OSProfile updates
+  2. Instance updates
+  3. New OSUpdatePolicy resource
+  4. New OSUpdateRun resource
+- **API**: API needs to be updated according to the schema changes.
+  1. OperatingSystem resource update
+  2. Instance resource update
+  3. New OSUpdatePolicy APIs (Create, Read, Delete)
+  4. New OSUpdateRun APIs (Read, Delete)
+- **Maintenance Manager (MM)**: The MM needs to be updated to handle the new 
+  OSUpdatePolicy and OSUpdateRun resources.
+  1. Update southbound APIs to allow PUA to provide packages ready to be 
+     updated
+  2. Handle OSUpdatePolicy
+  3. Create and Update OSUpdateRun
+- **OS Resource Manager (OSRM)**: OSRM adds support for new EMT versions, 
+  notice 
+  of new updates for EMT, and fill package info from Ubuntu manifest.
+  1. Polling of new EMT versions from Release Service and create new 
+     OSprofiles accordingly
+  2. Update `new_versions` field in the Instances
+  3. Fetch Ubuntu Manifest referenced into the OSProfile from Release 
+     Service. Parse it and fill the information into the OSProfile in 
+     Inventory.
+- **Host Resource Manager (HRM)**: HRM updates the installed packaged on 
+  per-instance basis
+  1. Update Southbound API to allow HDA to provide list of installed 
+     packages with versions.
+  2. Fill the installed packages information into the Instance resource.
+
+Edge Node:
+- **Platform Update Agent (PUA)**:
+  1. For mutable OSes, PUA needs to poll and check for new update to 
+     already installed packages, and reports back to the MM the new 
+     versions available.
+- **Hardware Discovery Agent (HDA)**:
+  1. For both mutable and immutable OSes, HDA should report the list of 
+     installed packages to the HRM with versions.
+
+UI/CLI:
+1. Support for creation of OSUpdatePolicy.
+2. Support to show History of the Instance, via the OSUpdateRun resources.
+3. Update to support the `new_versions` field in the Instance resource, 
+   instead of using `desired_os` to support update badge.
+
+## Implementation plan
+
+The implementation plan is divided into 3 phases.
+
+### Phase 1 (for EMF 3.1 release)
+
+1. schema and REST API changes (inv.\*, API.\*)
+2. Update to Maintenance Manager (MM) to handle OSUpdatePolicy, and create
+   and update OSUpdateRun (MM.ii, MM.iii)
+3. UI/CLI support for OSUpdatePolicy (UI/CLI.1) - UI work is Tentative
+4. CLI support for OSUpdateRun (CLI.2)
+5. UI update to support `new_versions` (UI/CLI.3) - Tentative
+5. New Integration tests
+6. Update to documentation for the new Day2 operations
+
+### Phase 2 (for EMF 3.2 release)
+
+1. OSRM to support immutable OSes notice for new versions (OSRM.i, OSRM.ii)
```

Comment:
does this mean that for phase 1 we will still create Profiles Manually for updated version ? please write this down if so.

Reply:
also phase 2 seems light, is it because you are planning CVE tracking in 3.2 ?

Reply:
anyway, refactor the plan a little bit. The most of the work is still on phase1, objective is to keep everything as much backward compatible as possible. Phase2 instead introduces refinements and removal of deprecated fields.